### PR TITLE
Show Juju commands in the changelog.

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3127.

Helps users learn the Juju command line, might also help a bit with #3314.

### CODE REVIEW

**Comment by huwshimi** [comment]
[Show Juju commands in the changelog](https://trello.com/c/c4t338wC/23-show-juju-commands-in-the-changelog)

**Comment by huwshimi** [comment]
QA:
- start a new model and add a bunch of things
- open the deployment flow
- in the change log the items should show the equivalent juju command

**Comment by jujugui** [comment]
Build finished.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8261/

File: jujugui/static/gui/src/app/init/changes-utils.js [hunk 1]
```diff
@@ -164,6 +164,22 @@ const changesUtils = {
   },
 
   /**
+    Convert an object to a keyval string e.g. {one: 'two', three: 'four'}
+    becomes one='two' three='four'.
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/changes-utils.js]: Please document what happens when the value is falsey.

File: jujugui/static/gui/src/app/init/changes-utils.js [hunk 2]
```diff
@@ -213,19 +231,24 @@ const changesUtils = {
           changeItem.icon = ghostService.get('icon');
           changeItem.description = ' ' + ghostService.get('name') +
               ' will be added to the model.';
+          changeItem.command = `juju deploy ${ghostService.get('charm')}`;
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/changes-utils.js]: Should we add the app name here? Or maybe only in the case it's different from the charm name?

File: jujugui/static/gui/src/app/init/changes-utils.js [hunk 3]
```diff
@@ -256,6 +283,8 @@ const changesUtils = {
           changeItem.description = change.command.args[0][1].name +
               ' relation will be added between ' +
               serviceList[0] + ' and ' + serviceList[1] + '.';
+          changeItem.command = `juju add-relation ${serviceList[0]} ` +
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/changes-utils.js]: Let's use "relate" instead?

File: jujugui/static/gui/src/app/init/changes-utils.js [hunk 4]
```diff
@@ -282,12 +325,19 @@ const changesUtils = {
...
+          ghostService = services.getById(change.command.args[0]);
+          const config = changesUtils._objectToParams(change.command.args[1]);
           changeItem.icon = 'changes-config-changed';
           changeItem.description = 'Configuration values will be changed for ' +
               cfgServ.get('displayName').match(removeBrackets)[1] + '.';
+          changeItem.command = `juju config ${ghostService.get('name')} ` +
+            `${config}`;
           break;
         case '_addKeys':
           changeItem.icon = 'changes-unknown';
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/changes-utils.js]: I don't see tests for all these commands: istm that only "juju deploy" is exercised by existing tests.

**Comment by frankban** [comment]
QA, as suspected, when setting only one constraint the resulting command is "juju add-machine -n 1 --constraints arch cpu-power=4 cpu-cores mem root-disk" which is not ok. Otherwise this seems to work well.

File: jujugui/static/gui/src/app/init/changes-utils.js (hunk 1 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/init/changes-utils.js]: Done.

File: jujugui/static/gui/src/app/init/changes-utils.js (hunk 3 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/init/changes-utils.js]: Done.

File: jujugui/static/gui/src/app/init/changes-utils.js (hunk 2 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/init/changes-utils.js]: Good point, done.

File: jujugui/static/gui/src/app/init/changes-utils.js (hunk 4 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/init/changes-utils.js]: Added.

**Comment by jujugui** [comment]
Build finished.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8262/

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**